tss-esapi-sys/build.rs

```diff
@@ -1,114 +1,487 @@
+                        .blocklist_type("IMAGE_TLS_DIRECTORY64")
+                        .blocklist_type("PIMAGE_TLS_DIRECTORY64")
+                        .blocklist_type("_IMAGE_TLS_DIRECTORY64")
+                        .blocklist_type("MONITORINFOEX")
+                        .blocklist_type("MONITORINFOEXA")
+                        .blocklist_type("MONITORINFOEXW")
+                        .blocklist_type("tagMONITORINFOEXA")
+                        .blocklist_type("tagMONITORINFOEXW")
+                        .blocklist_type("LPMONITORINFOEX")
+                        .blocklist_type("LPMONITORINFOEXA")
+                        .blocklist_type("LPMONITORINFOEXW")
+                        .derive_default(true);
+                    if let Some(tss2_tcti_tbs) = &self.tss2_tcti_tbs {
+                        builder = builder
+                            .clang_arg(tss2_tcti_tbs.include_dir_arg())
+                            .header(tss2_tcti_tbs.header_file_arg());
+                    }
+                    builder
+                }
+            }
+        }
+        /// Retrieves the installation path from the environment variable and validates it.
+        fn installation_path_from_env_var() -> Option<(PathBuf, String)> {
+            std::env::var(PATH_ENV_VAR_NAME).map_or_else(
+                |e| match e {
+                    std::env::VarError::NotUnicode(invalid_value) => {
+                        panic!(
+                            "Invalid `{}` env var: `{:?}`.",
+                            PATH_ENV_VAR_NAME, invalid_value
+                        );
+                    }
+                    std::env::VarError::NotPresent => None,
+                },
+                |var| Some(Installation::ensure_valid_installation_path(var)),
+            )
+        }
+
+        /// Ensures that the installation path is valid.
+        ///
+        /// # Details
+        /// In order to be considered valid the following
+        /// requirements needs to be full filled:
+        /// 1. The directory must exist.
+        /// 2. Sub directories `include` and `lib` must exist.
+        /// 3. A `VERSION` file must be present in the directory and it needs to be
+        ///    be specifying a version that is greater then the minimum supported version.
+        ///
+        /// The function is also responsible for reporting the library search path the rust compiler
+        /// should use.
+        ///
+        /// # Arguments
+        /// env_var - The value of the environment variable that contains the installation path.
+        ///
+        /// # Returns
+        /// A tuple containing the validated installation path and the version associated with it.
+        fn ensure_valid_installation_path(env_var: String) -> (PathBuf, String) {
+            let install_path = PathBuf::from(env_var);
+            if !install_path.is_dir() {
+                panic!(
+                    "`{}` specifies a path `{}`, that does not exist",
```

Comment:
Isn't this somewhat unrepresentative of the check before - if `env_var` points at a file instead of a dir, this panic will be triggered.

Reply:
Ah yes! Nice catch it should prompt an error saying that the installation path is expected to be a directory.

Reply:
Fixed

Reply:
Thanks!